magenta-lib/src/main/scala/magenta/input/resolver/package.scala

```diff
@@ -0,0 +1,9 @@
+package magenta.input
+
+import cats.data.Validated
+
+package object resolver {
+  implicit class RichValidated[E, A](validated: Validated[E, A]) {
+    def flatMap[EE >: E, B](f: A => Validated[EE, B]): Validated[EE, B] = validated.andThen(f)
```

Comment:
The reason it is here is to make for comprehensions work and eliminate the uses of andThen across the codebase.